inspirehep/dojson/hep/fields/bd1xx.py

```diff
@@ -197,27 +168,72 @@ def authors2marc(self, key, value):
     """Main Entry-Personal Name."""
     value = force_force_list(value)
 
-    def get_value(value):
-        affiliations = [
+    def _get_ids(value):
+        ids = {
+            'i': [],
+            'j': [],
+        }
+        if value.get('ids'):
+            for _id in value.get('ids'):
+                if _id.get('type') == 'INSPIRE ID':
+                    ids['i'].append(_id.get('value'))
+                elif _id.get('type') == 'ORCID':
+                    ids['j'].append('ORCID:' + _id.get('value')[6:])
+                elif _id.get('type') == 'JACOW':
+                    ids['j'].append('JACOW-' + _id.get('value')[5:])
+                elif _id.get('type') == 'CERN':
+                    ids['j'].append('CCID-' + _id.get('value')[5:])
+        return ids
+
+    def _get_affiliations(value):
+        return [
             aff.get('value') for aff in value.get('affiliations', [])
         ]
+
+    def _get_inspire_roles(value):
+        inspire_roles = []
+        if value.get('inspire_roles'):
+            for role in value.get('inspire_roles'):
+                if role == 'editor':
+                    inspire_roles.append('ed.')
+                else:
+                    inspire_roles.append(role)
+        return inspire_roles
+
+    def get_value_100_700(value):
+        ids = _get_ids(value)
         return {
             'a': value.get('full_name'),
-            'e': utils_get_value(value, 'contributor_roles.value'),
+            'e': _get_inspire_roles(value),
```

Comment:
This is wrong. The only allowed value here is `ed.` when `inspire_roles` contains `editor`.